bindings/node/src/mongocrypt.cc

```diff
@@ -528,64 +528,86 @@ Value MongoCrypt::MakeExplicitEncryptionContext(const CallbackInfo& info) {
-            }
+        if (!keyId.IsBuffer()) {
+            throw TypeError::New(Env(), "`keyId` must be a Buffer");
+        }
 
-            std::unique_ptr<mongocrypt_binary_t, MongoCryptBinaryDeleter> binary(BufferToBinary(keyId.As<Uint8Array>()));
-            if (!mongocrypt_ctx_setopt_key_id(context.get(), binary.get())) {
-                throw TypeError::New(Env(), errorStringFromStatus(context.get()));
-            }
+        std::unique_ptr<mongocrypt_binary_t, MongoCryptBinaryDeleter> binary(BufferToBinary(keyId.As<Uint8Array>()));
+        if (!mongocrypt_ctx_setopt_key_id(context.get(), binary.get())) {
+            throw TypeError::New(Env(), errorStringFromStatus(context.get()));
         }
+    }
 
-        if (options.Has("keyAltName")) {
-            Napi::Value keyAltName = options["keyAltName"];
+    if (options.Has("keyAltName")) {
+        Napi::Value keyAltName = options["keyAltName"];
 
-            if (!keyAltName.IsBuffer()) {
-                throw TypeError::New(Env(), "`keyAltName` must be a Buffer");
-            }
+        if (!keyAltName.IsBuffer()) {
+            throw TypeError::New(Env(), "`keyAltName` must be a Buffer");
+        }
 
-            std::unique_ptr<mongocrypt_binary_t, MongoCryptBinaryDeleter> binary(
-                BufferToBinary(keyAltName.As<Uint8Array>()));
-            if (!mongocrypt_ctx_setopt_key_alt_name(context.get(), binary.get())) {
-                throw TypeError::New(Env(), errorStringFromStatus(context.get()));
-            }
+        std::unique_ptr<mongocrypt_binary_t, MongoCryptBinaryDeleter> binary(
+            BufferToBinary(keyAltName.As<Uint8Array>()));
+        if (!mongocrypt_ctx_setopt_key_alt_name(context.get(), binary.get())) {
+            throw TypeError::New(Env(), errorStringFromStatus(context.get()));
         }
+    }
 
-        if (options.Has("algorithm")) {
-            std::string algorithm = options.Get("algorithm").ToString();
-            if (!mongocrypt_ctx_setopt_algorithm(
+    if (options.Has("algorithm")) {
+        std::string algorithm = options.Get("algorithm").ToString();
+        if (!mongocrypt_ctx_setopt_algorithm(
                     context.get(), algorithm.c_str(), algorithm.size())) {
-                throw TypeError::New(Env(), errorStringFromStatus(context.get()));
-            }
+            throw TypeError::New(Env(), errorStringFromStatus(context.get()));
         }
 
-        if (options.Has("contentionFactor")) {
-            Napi::Value contention_factor_value = options["contentionFactor"];
-            int64_t contention_factor = contention_factor_value.IsBigInt() ?
-                contention_factor_value.As<BigInt>().Int64Value(nullptr) :
-                contention_factor_value.ToNumber().Int64Value();
-            if (!mongocrypt_ctx_setopt_contention_factor(context.get(), contention_factor)) {
-                throw TypeError::New(Env(), errorStringFromStatus(context.get()));
+        if (algorithm == "rangepreview") {
```

Comment:
If you want to avoid the extra lower-case check on the JS side (and that might be a good idea, considering that now this logic is split into two separate locations), you can also use

```suggestion
        if (strcasecmp(algorithm.c_str(), "rangepreview") == 0) {
```

here for case-insentive comparison